src/main/java/de/tum/cit/aet/artemis/iris/web/open/PublicPyrisStatusUpdateResource.java

```diff
@@ -171,7 +183,58 @@ public ResponseEntity<Void> setStatusOfIngestionJob(@PathVariable String runId,
         }
 
         pyrisStatusUpdateService.handleStatusUpdate(ingestionWebhookJob, statusUpdateDTO);
-
         return ResponseEntity.ok().build();
     }
+
+    /**
+     * Retrieves the overall ingestion state of a lecture by communicating with Pyris.
+     *
+     * <p>
+     * This method sends a GET request to the external Pyris service to fetch the current ingestion
+     * state of all lectures in a course, identified by its `lectureId`. The ingestion state can be aggregated from
+     * multiple lecture units or can reflect the overall status of the lecture ingestion process.
+     * </p>
+     *
+     * @param courseId the ID of the lecture for which the ingestion state is being requested
+     * @return a {@link ResponseEntity} containing the {@link IngestionState} of the lecture,
+     */
+    @GetMapping("courses/{courseId}/lectures/ingestion-state")
+    @EnforceNothing
+    public ResponseEntity<Map<Long, IngestionState>> getStatusOfLectureIngestion(@PathVariable long courseId) {
+        try {
+            return ResponseEntity.ok(pyrisConnectorService.getLecturesIngestionState(courseId));
+        }
+        catch (Exception e) {
+            log.error("Error fetching ingestion state for course {}", courseId, e);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+        }
+    }
+
+    /**
+     * Retrieves the ingestion state of all lecture unit in a lecture by communicating with Pyris.
+     *
+     * <p>
+     * This method sends a GET request to the external Pyris service to fetch the current ingestion
+     * state of a lecture unit, identified by its ID. It constructs a request using the provided
+     * `lectureId` and `lectureUnitId` and returns the state of the ingestion process (e.g., NOT_STARTED,
+     * IN_PROGRESS, DONE, ERROR).
+     * </p>
+     *
+     * @param courseId  the ID of the lecture the unit belongs to
+     * @param lectureId the ID of the lecture the unit belongs to
+     * @return a {@link ResponseEntity} containing the {@link IngestionState} of the lecture unit,
+     */
+    @GetMapping("courses/{courseId}/lectures/{lectureId}/lecture-units/ingestion-state")
+    @EnforceNothing
+    public ResponseEntity<Map<Long, IngestionState>> getStatusOfLectureUnitsIngestion(@PathVariable long courseId, @PathVariable long lectureId) {
+
+        try {
+            return ResponseEntity.ok(pyrisConnectorService.getLectureUnitsIngestionState(courseId, lectureId));
+        }
+        catch (Exception e) {
+            log.error("Error fetching ingestion state for lecture {}", lectureId, e);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
```

Comment:
This is not a good way of error handling. I suggest you either suppress the error completely or you provide a reason to the client why the request failed. I prefer the second option, because this would make it easier to identify errors.